ompi/mca/osc/ucx/osc_ucx_component.c

```diff
@@ -432,10 +464,10 @@ static int component_select(struct ompi_win_t *win, void **base, size_t size, in
         }
     }
 
-    ret = opal_common_ucx_wpctx_create(mca_osc_ucx_component.wpool, comm_size,
-                                     &exchange_len_info, (void *)module->comm,
-                                     &module->ctx);
-    if (OMPI_SUCCESS != ret) {
+    /* Populate addr table */
+    ret = opal_common_ucx_wpool_update_addr(mca_osc_ucx_component.wpool, comm_size,
```

Comment:
Is there a way to check whether all addresses have been exchanged on that particular communicator and avoid the exchange after the first window has been allocated?